src/saveload/saveload.cpp

```diff
@@ -1501,6 +1559,34 @@ static inline bool SlIsObjectValidInSavegame(const SaveLoad &sld)
 	return (_sl_version >= sld.version_from && _sl_version < sld.version_to);
 }
 
+/**
+ * Calculate the size of the table header.
+ * @param slt The SaveLoad table with objects to save/load.
+ * @return size of given object.
+ */
+static size_t SlCalcTableHeader(const SaveLoadTable &slt)
+{
+	size_t length = 0;
+
+	for (auto &sld : slt) {
+		if (!SlIsObjectValidInSavegame(sld)) continue;
+
+		length += SlCalcConvFileLen(SLE_UINT8);
+		length += SlCalcStdStringLen(&sld.name);
+	}
+
+	length += SlCalcConvFileLen(SLE_UINT8);
```

Comment:
Based on the code alone I can't form an idea what this field in the table is going to do. So a comment would be appreciated. For the two in the loop before I'd guess one is the type and the other the name field, so that's at least somewhat clearer from the code itself.

Upon further reading it is the flag to denote end-of list, so maybe a `// End of field list`?